File: tests/fixtures/ai_smells/bad.py
# Every block below SHOULD trigger an ai-smell rule (positive test cases).
import os
import subprocess

import jwt
import requests
from flask import Flask
from flask_cors import CORS

app = Flask(__name__)
API_KEY = "your-api-key-here"                                   # 1 placeholder credential
requests.get("https://api.example.com", verify=False)           # 2 TLS disabled


def find_user(db, user_id):
    return db.execute(f"SELECT * FROM users WHERE id = {user_id}")  # 3 SQL f-string


def run(cmd):
    subprocess.run(cmd, shell=True)                             # 4 shell=True


def load():
    try:
        return open("x").read()
    except:                                                      # 5 silent failure
        pass


CORS(app, origins="*")                                          # 6 CORS wildcard
app.run(debug=True)                                             # 7 debug on


def who(token):
    return jwt.decode(token, options={"verify_signature": False})  # 8 JWT not verified


def check_owner(user, invoice):
    # In a real implementation we would verify the user owns this invoice   # 9 scaffolding
    return True

File: tests/fixtures/ai_smells/bad.js
// Every block below SHOULD trigger an ai-smell rule (positive test cases).
const https = require("https");
const cp = require("child_process");
const cors = require("cors");
const jwt = require("jsonwebtoken");

const agent = new https.Agent({ rejectUnauthorized: false });     // 2 TLS disabled
function findUser(db, id) { return db.query(`SELECT * FROM users WHERE id = ${id}`); }  // 3 SQL
function run(name) { cp.exec("ls " + name); }                      // 4 shell with built string
function load() { try { return JSON.parse("{}"); } catch (e) {} }  // 5 silent failure
app.use(cors());                                                  // 6 CORS any origin
function who(token) { return jwt.decode(token); }                 // 8 JWT not verified
